.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -144,18 +144,40 @@ private DateTimeResolutionResult ParseBasicRegexMatch(string text, DateObject re
             foreach (var regex in this.config.DateRegexes)
             {
                 var offset = 0;
+                var swift = -2;
                 var match = regex.Match(trimmedText);
                 if (!match.Success)
                 {
                     match = regex.Match(this.config.DateTokenPrefix + trimmedText);
-                    offset = this.config.DateTokenPrefix.Length;
+
+                    // Handing cases with "on"
```

Comment:
Why is this change needed if you've made extra changes to the base regex for this case?
Sorry, I can't get the meaning of the change just from the code.

Reply:
Only when the front is "on", "5.12/24-12" will be recognized as date by DateExtractor6/8. So if I don't change the basic regex, "on this 5.12", "5.12" will not be matched, we can't continue with the rest of the operation.